docs/design-documents/platform/pin-names/pin_names.md

```diff
@@ -0,0 +1,198 @@
+
+1. [Table of contents](#table-of-contents).
+    * [Revision history](#revision-history).
+1. [Introduction](#introduction).
+    1. [Overview and background](#overview-and-background).
+    1. [Requirements and assumptions](#requirements-and-assumptions).
+1. [Detailed design](#detailed-design).
+    1. [Arduino Uno (Rev 3) Header Pins](#arduino-uno-rev-3-header-pins).
+        1. [I2C Definition for Arduino Uno (Rev3) Pins](#i2c-definition-for-arduino-uno-rev3-pins)
+        1. [SPI Definition for Arduino Uno (Rev3) Pins](#spi-definition-for-arduino-uno-rev3-pins)
+    1. [Custom/Vendor Header Pins](#custom/vendor-header-pins).
+        1. [Custom/Vendor Connector I2C Definition](#custom/vendor-connector-i2c-definition).
+        1. [Custom/Vendor Connector SPI Definition](#custom/vendor-connector-spi-definition).
+    1. [Board Components Pins](#board-components-pins).
+        1. [LED Definition](#led-definition)
+        1. [BUTTON Definition](#button-definition)
+1. [Other information](#other-information).
+    1. [Avoid Unnecessary Definitions](#avoid-unnecessary-definitions).
+
+
+### Revision history
+
+1.0 - Initial revision - Malavika Sajikumar - 3/5/2020.  
+This document was written for MBED OS version 5.15 (released on 11th of February 2020).
+
+# Introduction
+
+### Overview and background
+
+Mbed is designed so that application code written in the platform is portable across different Mbed supported boards with the same hardware capabilities or interfaces. However, the code, in most cases, is not truly portable due to the differences in pin name definitions for the same kind of interfaces across different boards. 
+
+This document provides a standard for defining pin names for the commonly used Mbed interfaces like I2C, SPI, etc. and board components like buttons and LEDs in a board support package.
+
+### Requirements and assumptions
+
+This document mainly applies to the pin standards required for Arduino Uno header which is shared among multiple Mbed enabled boards. The Arduino Uno header has been stable since 2012 at its current revision, which is the Arduino Uno Rev3. All design choices in this document for the Arduino Uno header are based on the Arduino Uno Rev3 header implementation.
+
+# Detailed design
+
+To achieve meaningful portability of application code across various Mbed enabled boards, the pin names of commonly used interfaces and board components should be common across these boards. This document describes a set of rules on how to name these pins in the board support package, specifically in PinNames.h file of the board.
+
+### Arduino Uno (Rev3) Header Pins
+
+All Arduino Uno (Rev3) form factor Mbed controller boards should define D0-D15 & A0-A5 pins as a default standard.
+
+    // Arduino Uno (Rev3) header pin connection naming  
+    // Px_xx relates to the processor pin connected to the Arduino Uno (Rev3) header pin
+    
+    D0 = Px_xx,  // UART RX / GPIO 0  
+    D1 = Px_xx,  // UART TX / GPIO 1  
```

Comment:
I would remove comments for each lines...
Each pin can be more.

Reply:
@jeromecoutant If each pin is more, isn't that more reason to add comments?

Reply:
a comment describing alternative functionality for each pin? It should be stated in the description